### PR TITLE
Skip audit reports for ended, archived, test, or empty opportunities

### PR DESCRIPTION
**Pull request by hemant10yadav**
## Product Description

Weekly audit reports (and their email notifications) are no longer generated for opportunities that have ended, are archived, marked as test, or have no accepted workers.

## Technical Summary

Ticket: [CI-881](https://dimagi.atlassian.net/browse/CI-881)

- `generate_audit_report_for_opportunity` now returns `None` and skips creating a report when an opportunity has no active accesses, instead of creating an empty report with no entries.
- The task filters on `end_date__gte=period_start` rather than an "is active" flag, so an opportunity that ended partway through the just-completed week still gets its final report.

### Safety story

Ran the updated test suite locally and confirmed each exclusion case (ended, archived, test, no deliveries) is skipped while a normal opportunity and one ending mid-period still generate a report.

### Automated test coverage

Added tests for each exclusion case and the mid-period edge case, and updated the existing service/task tests for the new `None` return; all pass.

### QA Plan

 tested manually on local. Will request QA

### Deployment

- [x] This PR can be deployed: any required configuration, commands, or other manual steps required before this PR can be deployed are done.

### Labels & Review

- [ ] The set of people pinged as reviewers is appropriate for the level of risk of the change

[CI-881]: https://dimagi.atlassian.net/browse/CI-881?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: commcare_connect/audit/tasks.py [hunk 1]
```diff
@@ -35,6 +35,9 @@ def generate_audit_reports() -> None:
         Opportunity.objects.filter(
             Q(pk__in=flag.opportunities.values_list("pk", flat=True)) | Q(program__in=flag.programs.all()),
             active=True,
+            archived=False,
+            is_test=False,
+            end_date__gte=period_start,
```
Comment (mkangia) [inline commcare_connect/audit/tasks.py:40]: just confirming that this does not let to any weird situations where the end date for an opportunity is between the end of a period and the start of the next period?

File: commcare_connect/audit/services.py [hunk 2]
```diff
@@ -117,6 +111,15 @@ def generate_audit_report_for_opportunity(
         .order_by("user__name")
     )
 
+    if not active_accesses.exists():
```
Comment (mkangia) [inline commcare_connect/audit/services.py:114]: I assume this is additional to what the ticket was originally about?
So this now skips generating the report entirely, though the user does not really know why it was skipped?

File: commcare_connect/audit/tasks.py (hunk 1 above)
Comment (hemant10yadav) [reply commcare_connect/audit/tasks.py:40]: Yeah, I have considered that case as well, and it will produce a report for that too.

File: commcare_connect/audit/services.py (hunk 2 above)
Comment (hemant10yadav) [reply commcare_connect/audit/services.py:114]: If there are no users in the opportunity, there’s no point in generating a report or notifying the admins, right?

Comment (mkangia) [reply commcare_connect/audit/services.py:114]: Agreed.

I am just thinking from user perspective.

Are we expecting that report generation won't happen initially but then will always continue to happen since users once assigned will continue to exist and we won't have a scenario again where they won't be any accepted access until the opportunity finishes?

Comment (hemant10yadav) [reply commcare_connect/audit/services.py:114]: Yes

Comment (mkangia) [reply commcare_connect/audit/services.py:114]: okay :+1:

**Review by mkangia — APPROVED** [review]
